ykrt/src/compile/jitc_yk/mod.rs

```diff
@@ -0,0 +1,44 @@
+//! Yk's built-in trace compiler.
+
+use crate::{compile::Compiler, trace::MappedTrace};
+use libc::c_void;
+use std::{error::Error, ffi::CString, slice, sync::Arc};
+use tempfile::NamedTempFile;
+use ykaddr::addr::symbol_vaddr;
+
+mod aot_ir;
+
+pub(crate) struct JITCYk;
+
+impl Compiler for JITCYk {
+    fn compile(
+        &self,
+        _irtrace: MappedTrace,
```

Comment:
The interface for this has changed in master. You'll need to add:

```
        sti: Option<SideTraceInfo>,
        hl: Arc<Mutex<HotLocation>>,
```

I suggest `panic`ing if `sti.is_some()` at first.

Reply:
ok

Reply:
Merged upstream (and resolved conflicts) in [84e625e](https://github.com/ykjit/yk/pull/842/commits/84e625ec6d6b2878620fdf81cc684c8fd0ff8cf2), then added the proposed check in [6e47039](https://github.com/ykjit/yk/pull/842/commits/6e4703968ee53ca81dfc67a6f6fd8a94a62fe6da)